test/core.jl

```diff
@@ -4684,9 +4684,15 @@ end
     f18725(x) = 2
 end
 @test Main.f18725(0) == 2
-@test_warn "WARNING: Method definition f18725(Any) in module Module18725" @eval Main module Module18725
-    f18725(x) = 1
-    f18725(x) = 2
+
+# PR #23030
+@test_nowarn @eval Main module Module23030
```

Comment:
AFAIU, the previous version never defined a replacement for `Module18725` but checked that the replacement of the *function* `f18725` *within* `Module18725` gave a warning (since it was not in `Main`).